repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java

```diff
@@ -225,6 +227,14 @@ private void defineArgs() throws JSAPException {
         opt2.setHelp("Specify the build id to use.");
         this.jsap.registerParameter(opt2);
 
+        opt2 = new FlaggedOption("nextBuild");
+        opt2.setShortFlag('n');
+        opt2.setLongFlag("nextBuild");
+        opt2.setStringParser(JSAP.INTEGER_PARSER);
+        opt2.setDefault("0");
+        opt2.setHelp("Specify the next build id to use.");
```

Comment:
You should specify that it is mandatory when in bears mode and maybe add a check to enforce that.

Reply:
Not really. See the changes in the method `getBuildToBeInspected`. I relax the approach by now (not sure if I will keep this way, I will see), but I record in the logs that the next build was retrieved with jTravis, not given as input.

Reply:
Hmmm. I'm not sure to follow you on this one. I think it's better to put it as a mandatory argument for Bears: you should have get a pair of build ids before calling the pipeline, so you should be able to put it in argument.
Do you see a usecase where a user cannot give it as argument?

Reply:
I understand your thought. I'm really divided between leaving it relaxed or not. Let me tell you first why I'm doing this pr in the first place: do you remember that you sent me a list of 10 branches from bugs-collection that are outside the time window we looked for bugs, and there was a specific branch that was a good one, but I didn't know why it was outside the time window? I strongly believe it was a problem of a build being removed. In other words, in the scanner, the buggy build was linked with a fixer build, and the latter was removed. Then, in the pipeline, jTravis returned a different fixer build (which was outside the time window) for that buggy build. See that in practice there is no problem as the pipeline makes sure that the buggy builds need to fail and the fixer builds need to pass, so thus we got a bug, even that the fixer build was not the same one we scan. All of this is what I *think*, but I couldn't confirm as we lost logs. However, that was the only explanation for that behaviour. Since I just gave you a context now on why I believe we won't get branches that are not bugs by relaxing the entry to the pipeline, I will go back to your question. I'm not sure about making it mandatory because I believe one usecase where a user cannot give it as argument is when the user used repairnator in repair mode, so she got a list of failing builds, but then she wants a collection of bugs containing also fixer builds. If I put the entry for Bears mandatory, she will need to run the scanner again to get a list of buggy and fixer build ids to give as input to the pipeline. This is the reason I thought about relaxing the entry to the pipeline, i.e., I think it can have usecase, and I don't think it can be a problem (it won't result in wrong stuff). What do you think about all of this? Does that make sense? If you tell me that this usecase won't make sense to happen, I will be happy to put the argument as mandatory.

Reply:
Ok there's many different things here! 

So first, concerning the question of making the arguments mandatory or not: I maintain it would be better.
Because: 
  1. you provide a _potential_ usecase that WE do not have, 
  2. it's not the concern of pipeline to compute pair of builds, so even if 1 is valid, I would say to use the scanner to get the pair of builds: that's its role

Then it's not really the place but when you say:
>  the buggy build was linked with a fixer build, and the latter was removed

It's just not possible to remove a build from Travis. Maybe you meant that the build would have been played again and its date changed? It might be possible.

Reply:
On turning the arg `nextBuild` mandatory for Bears, ok, I do it.

On the other thing, I don't know, and I can't be sure on what happened as I don't know with which fixer build that buggy build was linked in the scanner. I was just explaining why I started with all this thing on passing the fixer build id to the pipeline.